File: Summation of Primes/SummationOfPrimes.cpp
#include <iostream>
#include <vector>
using namespace std;

// function to verify if number is prime
bool primeNumber(int number)
{
    int numberHalf = number / 2;
    int divisor = 1;
    std::vector<int> divisorList;

    while (divisor <= numberHalf)
    {
        if (number % divisor == 0)
        {
            divisorList.push_back(divisor);
        }
        divisor = divisor + 1;
    }

    if (divisorList.size() == 1)
    {
        return true;
    }
    else
    {
        return false;
    }
}

int SummationOfPrimes(int number)
{
    int count = 1;
    vector<int> listCount;
    int summation = 0;

    while (count < number)
    {
        if (primeNumber(count) == true)
        {
            listCount.push_back(count);
            summation = summation + summation;
        }
        count = count + 1;
    }

    return summation;
}

/*int main()
{
    int number;
    cout << "Number: ";
    cin >> number;

    // challenge asks for 2000000
    cout << SummationOfPrimes(number);

    return 0;
}*/

File: Summation of Primes/SummationOfPrimes.py
def primeNumber(number):
    numberHalf = number / 2
    divisor = 1
    divisorList = []

    while divisor <= numberHalf:
        if number % divisor == 0:
            divisorList.append(divisor)
        divisor = divisor + 1

    if len(divisorList) == 1:
        return True
    else:
        return False


def SummationOfPrimes(number):
    count = 1
    listCount = []
    summation = 0

    while count < number:
        if primeNumber(count) == True:
            listCount.append(count)
            summation = summation + count
        count = count + 1
    print(f"{len(listCount)}, {listCount}.")

    return summation


number = int(input("Number: "))

# challenge asks for 2000000
print(SummationOfPrimes(number))
